packages/rule-no-broken-links/tests/no-broken-links.ts

```diff
@@ -0,0 +1,131 @@
+        serverConfig: {
+            '/': {content: generateHTMLPage('', bodyWithValidLinks)},
+            '/about': {content: 'My about page content'}
+        }
+    },
+    {
+        name: `This test should pass as it has an img with valid src value(absolute)`,
+        serverConfig: generateHTMLPage('', bodyWithImageSource)
+    },
+    {
+        name: `This test should fail as it has a link with 404 href value(absolute)`,
+        reports: [{ message: `Broken link found (404 response)` }],
+        serverConfig: generateHTMLPage('', bodyWithBrokenLinks)
+    },
+    {
+        name: `This test should fail as it has an img with 404 src value(absolute)`,
+        reports: [{ message: `Broken link found (404 response)` }],
+        serverConfig: generateHTMLPage('', bodyWithBrokenImageSource)
+    },
+    {
+        name: `This test should fail as it has a link with 404 href value(absolute)`,
+        reports: [{ message: `Broken link found (404 response)` }],
+        serverConfig: generateHTMLPage('', bodyWithValidLinksAndBrokenLinks)
+    },
+    {
+        name: `This test should fail as it has a link with 500 href value(relative)`,
+        reports: [{ message: `Broken link found (500 response)` }],
+        serverConfig: {
+            '/': {content: generateHTMLPage('', bodyWithRelative500Links)},
+            '/500': {status: 500}
+        }
+    },
+    {
+        name: `This test should fail as it has a link with 410 href value(relative)`,
+        reports: [{ message: `Broken link found (410 response)` }],
+        serverConfig: {
+            '/': {content: generateHTMLPage('', bodyWithRelative410Links)},
+            '/410': {status: 410}
+        }
+    },
+    {
+        name: `This test should fail as it has a link with 404 href value(relative)`,
+        reports: [{ message: `Broken link found (404 response)` }],
+        serverConfig: {
+            '/': {content: generateHTMLPage('', bodyWithRelative404Links)},
+            '/404': {status: 404}
+        }
+    },
+    {
+        name: `This test should fail as it has a link with 503 href value(relative)`,
+        reports: [{ message: `Broken link found (503 response)` }],
+        serverConfig: {
+            '/': {content: generateHTMLPage('', bodyWithRelative503Links)},
+            '/503': {status: 503}
+        }
+    },
+    {
+        name: `This test should fail as it has a link with 404 href value`,
+        reports: [{ message: `Broken link found (404 response)` }],
+        serverConfig: {
+            '/': {content: generateHTMLPage('', bodyWithBrokenScriptTag)},
+            '/404': {status: 404}
+        }
+    },
+    {
+        name: `This test should fail as it has a script with 404 src value`,
+        reports: [{ message: `Broken link found (404 response)` }],
+        serverConfig: {
+            '/': {content: generateHTMLPage('', bodyWithBrokenLinkTag)},
+            '/404': {status: 404}
+        }
+    }
```

Comment:
@kshyju The coverage is lower the 80%, can you add a tests for something such as: `<img alt="test" src="1.jpg" srcset="2.jpg 640w,3.jpg 750w , 4.jpg 1080w">`? Thanks!

Reply:
My bad :( My assertion statements were wrong. All good now. The rule has 85 % coverage now.